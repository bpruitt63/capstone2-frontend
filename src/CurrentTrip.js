import React from 'react';
import {Button} from 'reactstrap';
import './static/styles/Trips.css';
import TripNameForm from './TripNameForm';
import {calculateDistance} from './static/helpers';

/** Displays list of destinations as user creates a trip, 
 * followed by the total distance of the trip in either 
 * miles or kilometers, depending on selected units.
 * Includes button to completely remove trip.
 */

function CurrentTrip({currentTrip, deleteCurrentTrip, units, username, addTrip}) {

    /** Displays distance for trip.  
     * Takes distance from last location in currentTrip array
     */
    let totalDistance = 0;
    if (currentTrip[0]) {
        totalDistance = currentTrip[currentTrip.length - 1].totalDistance;
        totalDistance = calculateDistance(units, totalDistance);
    };
    
    return(
        <div className='tripsGroup'>
            <h3 className='pointsTitle'>Current Trip</h3>
            {!currentTrip[0] && 
                <p>No trip destinations selected</p>
            }
            {currentTrip.map(p => 
                <p key={p.totalDistance}>{p.name}</p>
            )}
            {currentTrip[0] && 
                <p>
                    Total Distance: {totalDistance} 
                    {units === 'i' ? ' mi' : ' km'}
                </p>
            }
            {currentTrip[1] &&
                <TripNameForm currentTrip={currentTrip}
                                username={username}
                                deleteCurrentTrip={deleteCurrentTrip}
                                addTrip={addTrip} />
            }
            <br/>
            {currentTrip[0] && 
                <Button onClick={deleteCurrentTrip} 
                        className='toggle'>
                    Delete Current Trip
                </Button>
            }
        </div>
    );
};

export default CurrentTrip;
