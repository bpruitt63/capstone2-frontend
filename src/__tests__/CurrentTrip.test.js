import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CurrentTrip from '../CurrentTrip';


test('renders', () => {
    render(<CurrentTrip currentTrip={[]}/>);
});

test('matches snapshot', () => {
    const {asFragment} = render(<CurrentTrip currentTrip={[]}/>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders w/ trip', () => {
    render(<CurrentTrip currentTrip={[{name: "test", 
                                        location:[20, 25], 
                                        id: 'testId',
                                        totalDistance: 0},
                                    {name: "test2", 
                                        location:[30, 35], 
                                        id: 'testId2',
                                        totalDistance: 33}]} />);
    expect(screen.getByText(/test2/)).toBeInTheDocument;
    expect(screen.getByText(/0.03/)).toBeInTheDocument;
});

test('deletes trip', () => {
    render(<CurrentTrip currentTrip={[{name: "test", 
                                        location:[20, 25], 
                                        id: 'testId',
                                        totalDistance: 0},
                                    {name: "test2", 
                                        location:[30, 35], 
                                        id: 'testId2',
                                        totalDistance: 33}]} />);
    const deleteButton = screen.getByText(/Delete Current Trip/);
    fireEvent.click(deleteButton);
    expect(screen.getByText(/test2/)).not.toBeInTheDocument;
    expect(screen.getByText(/0.03/)).not.toBeInTheDocument;
});

test('tripRatingForm shows only with trip', () => {
    render(<CurrentTrip currentTrip={[{name: "test", 
                                        location:[20, 25], 
                                        id: 'testId',
                                        totalDistance: 0},
                                    {name: "test2", 
                                        location:[30, 35], 
                                        id: 'testId2',
                                        totalDistance: 33}]} />);
    expect(screen.getByText(/Save Trip/)).toBeInTheDocument;
    const deleteButton = screen.getByText(/Delete Current Trip/);
    fireEvent.click(deleteButton);
    expect(screen.queryByText(/Save Trip/)).not.toBeInTheDocument;
});