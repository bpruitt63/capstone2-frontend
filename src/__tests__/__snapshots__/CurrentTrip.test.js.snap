// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tripsGroup"
  >
    <h3
      class="pointsTitle"
    >
      Current Trip
    </h3>
    <p>
      No trip destinations selected
    </p>
    <br />
  </div>
</DocumentFragment>
`;
